mmv1/products/firestore/UserCreds.yaml

```diff
@@ -0,0 +1,99 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'UserCreds'
+description: |
+  User credentials for a Cloud Firestore with MongoDB compatibility database.
+  The resource is owned by the database and is deleted along with the database.
+references:
+  guides:
+    'Authenticate and connect to a database': 'https://cloud.google.com/firestore/mongodb-compatibility/docs/connect'
+  api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.userCreds'
+base_url: 'projects/{{project}}/databases/{{database}}/userCreds'
+self_link: 'projects/{{project}}/databases/{{database}}/userCreds/{{name}}'
+immutable: true
+create_url: 'projects/{{project}}/databases/{{database}}/userCreds?userCredsId={{name}}'
+autogen_async: false
+custom_code:
+  post_create: templates/terraform/post_create/firestore_user_creds.go.tmpl
+examples:
+  - name: 'firestore_user_creds_basic'
+    primary_resource_id: 'my-user-creds'
+    vars:
+      database_id: 'database-id-mongodb-compatible'
+    test_env_vars:
+      project_id: 'PROJECT_NAME'
+  - name: 'firestore_user_creds_with_secret_manager'
+    primary_resource_id: 'my-user-creds'
+    vars:
+      database_id: 'database-id-mongodb-compatible'
+    test_env_vars:
+      project_id: 'PROJECT_NAME'
+parameters:
+properties:
+  - name: 'database'
+    type: String
+    description: |
+      The Firestore database ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - name: 'name'
+    type: String
+    description: |
+      The ID to use for the user creds, which will become the final component
+      of the user cred's resource name.
+
+      This value should be 4-63 characters. Valid characters are /[a-z][0-9]-/
+      with first character a letter and the last a letter or a number. Must not
+      be UUID-like /[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}/.
+    required: true
+    immutable: true
+    custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.tmpl'
+    custom_expand: 'templates/terraform/custom_expand/shortname_to_url.go.tmpl'
+  - name: 'create_time'
+    type: String
+    description: |
+      Output only. The timestamp at which these user creds were created.
+    output: true
+  - name: 'update_time'
+    type: String
+    description: |
+      Output only. The timestamp at which these user creds were updated.
+    output: true
+  - name: 'state'
+    type: String
+    description: |
+      Output only. The state of the user creds.
+    output: true
+  - name: 'secure_password'
```

Comment:
this field is a good candidate for a `write-only` field, can you address the requests first and attempt to make this a write-only field?

Reply:
The underlying REST API does not permit writes to this field (instead, the password is generated on the server and returned only once in response to the Create() call).

Reply:
yes as a write-only attribute it can still be treated as a sensitive type, however the difference is that a sensitive attribute still gets stored in state where as a write-only is not stored but still used when called.

Reply:
The problem is that our REST API does not allow the client to choose a value for this field (it can only be generated by the server):
https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.userCreds

Since the field cannot be written by the client, I don't think it can be `write_only`? (I may be misunderstanding)